VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionCommentsFragment.java

```diff
@@ -40,18 +42,15 @@
     private DiscussionComment discussionComment;
 
     @Inject
-    DiscussionAPI discussionAPI;
-
-    @Inject
-    DiscussionCommentsAdapter discussionCommentsAdapter;
-
-    @Inject
     Router router;
 
     @Inject
     Context context;
 
-    private static final Logger logger = new Logger(CourseDiscussionCommentsFragment.class.getName());
+    private DiscussionCommentsAdapter discussionCommentsAdapter;
+
+    @Nullable
```

Comment:
I'm not sure what you mean. This can definitely be analysed statically, and at the very least it will show IDE warnings if you try to access it without checking for null first.